pkg/cloudmap/utils.go

```diff
@@ -24,3 +37,69 @@ func ShouldPodBeInEndpoints(pod *corev1.Pod) bool {
 		return true
 	}
 }
+
+func AwaitOperationSuccess(timeoutSeconds time.Duration, tickerSeconds time.Duration, f func() string) string {
+	operationStatus := make(chan string, 1)
```

Comment:
This should just be something like 

endTime = wallClock + timeoutSeconds

while (wallClock < endTime) { 
   checkStatus();
   if (state == GOOD || FAIL) { 
      return;
   }
   return TIMEOUT;
}

Reply:
Also make sure you control Client timeout. In other words, client shouldn't go out and use a 30 second timeout for every attempt.

Reply:
Also make sure you control Client timeout. In other words, client shouldn't go out and use a 30 second timeout for every attempt.